DI-832; add maven enforcer checks for legacy artifacts

skuehn opened the pull request.

aaronphysics commented on pom.xml:
```diff
@@ -478,6 +478,42 @@
+                                        data-infrastructure@opower.com if you have questions</message>
+                                    <excludes>
+                                        <!-- hdfs -->
+                                        <exclude>org.apache.hadoop:hadoop-common:[2.0.0-cdh4.4.0]</exclude>
```
Can we cast a larger net with wildcards like `[2.0.0-cdh4.*.*]`?

skuehn replied:
Good question - version strings use range matching rules, which are different than the standard wildcards: [enforcer plugin version range ref](https://maven.apache.org/enforcer/enforcer-rules/versionRanges.html).

I'll try some of the version range matching rules to see if we can catch more offending versions.

skuehn commented:
@aaronphysics some improvements:
1. The version matchers now reject any version up to the target cdh version, rather than the version that is currently in use.
2. Added a rule to reject the hbase artifact. The cdh5 upgrade introduces `hbase-client` and `hbase-server`, so we'll fail fast if there's a build using plain `hbase`.
3. Fixed a typo with the hadoop-hdfs rule
